### PR TITLE
Add support for reservation affinity under dataproc cluster 

### PR DESCRIPTION
**Pull request by prateek2408**
Currently dataproc cluster  does not support reservation affinity under cluster config  . This PR adds the support of creating cluster config with resource affinity config.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
dataproc: added support for `"Resource Affinity under Cluster Config"` in `google_dataproc_cluster`

```

### CODE REVIEW

**Review by slevenick — CHANGES_REQUESTED** [review]
Have you tested this change to make sure it works?

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb [hunk 1]
```diff
@@ -638,6 +638,35 @@ func resourceDataprocCluster() *schema.Resource {
...
+									"reservation_affinity": {
+										Type:         schema.TypeList,
+										Optional:     true,
+										AtLeastOneOf: gceClusterConfigKeys,
+										Computed:     true,
+										MaxItems:     1,
+										Description:  `Reservation Affinity for consuming Zonal reservation.`,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"consume_reservation_tyoe": {
```
Comment (cjac) [inline mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:650]: should this be consume_reservation_type?

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb [hunk 2]
```diff
@@ -638,6 +638,35 @@ func resourceDataprocCluster() *schema.Resource {
...
+													Type:         schema.TypeEnum,
+													Optional:     true,
+													Default:      false,
+													AtLeastOneOf: schieldedInstanceConfigKeys,
```
Comment (cjac) [inline mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:654]: should this be shieldedInstanceConfigKeys ?

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb [hunk 3]
```diff
@@ -638,6 +638,35 @@ func resourceDataprocCluster() *schema.Resource {
...
+													ForceNew:     true,
+													Description:  `Defines whether instances have Secure Boot enabled.`,
+												},
+												"key": {
+													Type:         schema.String,
+													Optional:     true,
+													Default:      '',
+													AtLeastOneOf: schieldedInstanceConfigKeys,
```
Comment (cjac) [inline mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:662]: should this be shieldedInstanceConfigKeys ?

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb (hunk 2 above)
Comment (prateek2408) [reply mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:654]: It should be but the variable definition itself at ln 68 has the name

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb (hunk 3 above)
Comment (prateek2408) [reply mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:662]: It should be but the variable definition itself at ln 68 has the name

**Review by slevenick — CHANGES_REQUESTED** [review]
Adding fields to the schema is not sufficient to add functionality to the provider for a handwritten resource. Create/read/update functionality will need to be written as well.

Additionally we will need tests on these fields

**Comment by cjac** [comment]
Duplicate of #6889

**Pull request closed without merging**